### PR TITLE
create all classes and relation between them

### PR DESCRIPTION
**Pull request by ahmedtaa**
# In this branch:
- Create a class **Classroom**, **Book**, **Rental** 
- Create the `has-many/belongs-to` relationship between Classroom and Student
- Create the `many-to-many` (also has-many-through) relationship between Person and Book

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Ahmed, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have a vaild/ passing linter checks.
Kindly ensure that all linters are passing and have green checkmark(s) accordingly. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: student.rb [hunk 1]
```diff
@@ -1,12 +1,13 @@
 require './person'
 
 class Student < Person
```
Comment (rloterh) [inline student.rb:5]: - [x] OPTIONAL
Kindly mark _classroom_ as an `attr_reader`, so that it can be accessed and updated

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (rloterh) [inline book.rb]: - [x] We need both   _Getters_ and _Setters_ for _@rentals_ to be present, therefore kindly ensure to mark _rentals_ with _attr_accessor_, like:
```ruby
  attr_accessor :rentals

File: person.rb [hunk 3]
```diff
@@ -2,14 +2,15 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals, :parent_permission
```
Comment (rloterh) [inline person.rb]: - [x] We need both   _Getters_ and _Setters_ for _@rentals_ to be present, therefore kindly ensure to mark _rentals_ with _attr_accessor_, like:
```ruby
  attr_accessor :rentals

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Ahmed, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- Descriptive Pull Request
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly check the comments marked **[OPTIONAL]** under the review._
<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — APPROVED** [review]
# Hi @ahmedtaa 👍, 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**